### PR TITLE
fix(pdf): ignore phantom gengar skill text nodes

### PR DESCRIPTION
**Pull request by santhiprakash**
## Problem

Activating Semantic CSS on a `gengar` resume with populated `skills.keywords` fails parity because the legacy renderer emits a harmless empty `TEXT` node with `fontSize: 9`, and the parity comparator treats that artifact as a mismatch. See issue #3285.

## Triage / Root cause

- `packages/pdf/src/templates/shared/sections.tsx:1157` renders skills keywords via `item.keywords.join(", ")`.
- `packages/pdf/src/semantic/legacy-parity.ts` normalized some empty text artifacts, but it still rejected the specific empty `TEXT` node pattern produced by the legacy `gengar` skills rendering path.

## Fix

- Treat the reported phantom empty `TEXT` node at `fontSize: 9` as presentation-neutral in the legacy parity collector.
- Add a regression test that covers the gengar-style empty node while keeping the existing tampering guards intact.

## Verification

- `pnpm test src/semantic/legacy-parity.test.ts` in `packages/pdf` passed.
- Result: `31 passed`.

## Notes / Risks

- The normalization is intentionally narrow: it only covers the specific empty legacy text artifact reported in #3285 and does not relax non-empty text mismatch detection.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved PDF character sheet rendering by hiding empty or whitespace-only skill proficiency text.
  * Preserved proficiency details when valid text is available, ensuring skills remain displayed correctly.
  * Improved consistency for inherited text sizing in skill sections.

<details open><summary><h3>Greptile Summary</h3></summary>

This PR fixes a phantom empty `TEXT` node emitted by the gengar template's `SkillsSection` when `item.proficiency` is blank, which was causing false parity mismatches when Semantic CSS was activated.

- **`sections.tsx`**: Wraps the proficiency `<Text>` in a `hasSplitRowText` guard, matching the pattern already used for `location`, `period`, `position`, and `phone` throughout the file. This is the root-cause fix — nothing is emitted when proficiency is empty or whitespace-only.
- **`legacy-parity.test.ts`**: Adds a renderer-level regression test using the same `instance.container.document` polling pattern already present in `legacy-parity.ts`, verifying both that empty/whitespace proficiency produces no fontSize-9 text nodes and that a populated proficiency value (`"Expert"`) is preserved in the rendered output.
</details>

<details open><summary><h3>Confidence Score: 5/5</h3></summary>

- Safe to merge. The fix is a one-line conditional guard that matches an established pattern used consistently throughout the same file, and the accompanying regression test exercises the renderer directly.
- The change is minimal and narrowly scoped: one conditional guard on the proficiency `<Text>` node, backed by a new renderer-level regression test that exercises both the empty and non-empty proficiency paths. The `hasSplitRowText` helper is well-tested separately and already in wide use across `sections.tsx`. No data flows, schemas, or parity logic were altered.
- No files require special attention.
</details>

<details open><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| packages/pdf/src/templates/shared/sections.tsx | Adds a `hasSplitRowText` guard around the proficiency `<Text>` in `SkillsSection`, preventing an empty TEXT node from being emitted into the PDF document tree when `item.proficiency` is empty or whitespace-only. Change is narrowly scoped and consistent with the same guard used for `location`, `period`, `position`, and `phone` throughout the file. |
| packages/pdf/src/semantic/legacy-parity.test.ts | Adds a renderer-level regression test for the gengar empty proficiency phantom node, with helper functions (`renderHostDocument`, `findEmptyTextNodes`, `hasFontSize`) that mirror the production `waitForDocument` pattern already used in `legacy-parity.ts`. Also adds `pdf` and `createElement` imports to support the new async rendering path. |

</details>

<details open><summary><h3>Flowchart</h3></summary>

```mermaid
%%{init: {'theme': 'neutral'}}%%
flowchart TD
    A[SkillsSection renders skill item] --> B{hasSplitRowText\nitem.proficiency}
    B -->|false: empty or whitespace| C[Skip proficiency Text node\nNo phantom node emitted]
    B -->|true: non-empty string| D[Render Text node\nwith proficiency value]
    C --> E[Render Small node\nfor keywords]
    D --> E
    E --> F[PDF document tree]
    F --> G{Legacy parity check}
    G -->|No phantom empty TEXT node| H[Parity passes ✓]
```
</details>

<sub>Reviews (5): Last reviewed commit: ["Merge branch &#39;fix/gengar-empty-text-pari..."](https://github.com/amruthpillai/reactive-resume/commit/09744be828a630cf324a785961771891f0c9f6cb) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=49280179)</sub>

### CODE REVIEW

File: AGENTS.md [hunk 1]
```diff
@@ -115,6 +115,8 @@ Copy `.env.example` to `.env`. The three required variables are:
 
 S3/SeaweedFS is optional. If `S3_ACCESS_KEY_ID`, `S3_SECRET_ACCESS_KEY`, and `S3_BUCKET` are all set, the app uses S3-compatible storage. The checked-in `.env.example` sets SeaweedFS defaults, so either start the `seaweedfs` compose service too or comment out those S3 vars to use local filesystem storage under `<workspace>/data`. `LOCAL_STORAGE_PATH` must be absolute when set.
 
+`REDIS_URL` and `ENCRYPTION_SECRET` are optional for core resume flows, but both are required for saved AI providers and the authenticated `/agent` workspace. Start the `redis` compose service and set both vars in `.env.local` when working on those features.
+
```
[the remark this answers is not in the document]
Comment (amruthpillai) [reply AGENTS.md]: Not carried forward: this repair restores unrelated Redis documentation to origin/main. f8a351101.

File: packages/pdf/src/semantic/legacy-parity.ts [hunk 2]
```diff
@@ -65,6 +65,7 @@ const emptyTextResetStyle: Readonly<Record<string, unknown>> = {
 	maxWidth: "100%",
 	minWidth: 0,
 };
+const emptyTextPresentationFontSizes = new Set([9]);
```
[the remark this answers is not in the document]
Comment (amruthpillai) [reply packages/pdf/src/semantic/legacy-parity.ts]: Fixed at the renderer source and removed the comparator exception in f8a351101.

File: packages/pdf/src/semantic/legacy-parity.test.ts [hunk 3]
```diff
@@ -292,6 +292,19 @@ describe("compareLegacySemanticPresentation", () => {
...
+				{
+					type: "VIEW",
+					children: [
+						{ type: "TEXT", style: { fontSize: 9 }, value: "" },
+						{ type: "TEXT", value: "same" },
+					],
+				},
+				{ type: "VIEW", children: [{ type: "TEXT", value: "same" }] },
+			),
+		).toEqual([]);
```
[the remark this answers is not in the document]
Comment (amruthpillai) [reply packages/pdf/src/semantic/legacy-parity.test.ts]: Replaced the comparator-only test with a real Gengar renderer regression in f8a351101.

File: packages/pdf/src/semantic/legacy-parity.test.ts [hunk 4]
```diff
@@ -157,6 +160,27 @@ const buildFixture = (rules: StyleRule[]): ResumeData => {
...
+	const element = createElement(ResumeDocument, { data, template: "gengar" }) as unknown as Parameters<typeof pdf>[0];
+	const instance = pdf(element);
+	await expect.poll(() => instance.container.document).not.toBeNull();
+	return instance.container.document as LegacyParityHostNode;
+};
+
+const findEmptyTextNodes = (node: LegacyParityHostNode): LegacyParityHostNode[] => [
+	...(node.type === "TEXT" && nodeText(node) === "" ? [node] : []),
+	...(node.children ?? []).flatMap(findEmptyTextNodes),
+];
```
[the remark this answers is not in the document]
Comment (amruthpillai) [reply packages/pdf/src/semantic/legacy-parity.test.ts:176]: Covered empty and whitespace-only proficiency values; focused renderer test, PDF typecheck, and Biome passed in 696e997b7.

**Comment by amruthpillai** [comment]
Thank you for your contribution — I appreciate the time and care you put into this.
